lib/proxy.js

```diff
@@ -137,12 +137,13 @@ function buildResultingMiddleware (host, options, stream) {
     var reqOpts = {
       url     :  URL.format(proxyUrl),
       method  : req.method,
-      headers : _.defaults(options.headers || {}, req.headers),
+      // headers : _.defaults(options.headers || {}, req.headers),
+      headers : options.headers,
```

Comment:
I'm a little confused about how this is breaking supertest. shouldnt `reqOpts` be scoped to  `proxyMiddleware`, which is called on each request.

as for your last comment. I definitely understand that viewpoint. I just think allowing a user to opt in to a partial override makes life easier since all you need to do to override headers is just to dunp them in the `pre` request hook. We could also add an `ignoreReqHeaders` option in the config. Or something along those lines